### ISSUE TITLE
[BUG] No Operator Hub Items Found 

### ISSUE THREAD

**Issue by GrooveCS** [code-ready/crc#1313]
### General information

  * OS: **Windows**
  * Hypervisor: **Hyper-V** 
  * Did you run `crc setup` before starting it (Yes/No)? **YES**
  * Running CRC on: **Laptop**

## CRC version
```bash
# Put `crc version` output here
crc version: 1.11.0+883ca49
OpenShift version: 4.4.5 (embedded in binary)
```
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
DEBU Running 'C:\Users\Developer\.crc\bin\oc\oc.exe get co -ojson --kubeconfig C:\Users\Developer\.crc\machines\crc\kubeconfig --context admin --cluster crc'
DEBU About to run SSH command:
df -B1 --output=size,used,target /sysroot | tail -1
DEBU SSH cmd err, output: <nil>: 32719437824 12208078848 /sysroot
CRC VM:          Running
OpenShift:       Running (v4.4.5)
Disk Usage:      12.21GB of 32.72GB (Inside the CRC VM)
Cache Usage:     40.95GB
Cache Directory: C:\Users\Developer\.crc\cache
```

## CRC config
```bash
# Put `crc config view` output here
```

## Host Operating System
```bash
# Put the output of `systeminfo` in case of Windows

Host Name:                 DESKTOP-TF4L8CF
OS Name:                   Microsoft Windows 10 Pro for Workstations
OS Version:                10.0.18363 N/A Build 18363
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Developer
Registered Organization:   
Product ID:                00391-90050-00000-AA002
Original Install Date:     3/26/2020, 3:23:44 PM
System Boot Time:          6/12/2020, 5:19:30 PM
System Manufacturer:       Acer
System Model:              Aspire TC-865
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 158 Stepping 10 GenuineIntel ~2808 Mhz
BIOS Version:              American Megatrends Inc R01-A3, 5/16/2018
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume3
System Locale:             en-us;English (United States)
Input Locale:              N/A
Time Zone:                 (UTC-05:00) Eastern Time (US & Canada)
Total Physical Memory:     65,423 MB
Available Physical Memory: 44,535 MB
Virtual Memory: Max Size:  75,151 MB
Virtual Memory: Available: 53,862 MB
Virtual Memory: In Use:    21,289 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\DESKTOP-TF4L8CF
Hotfix(s):                 11 Hotfix(s) Installed.
                           [01]: KB4552931
                           [02]: KB4513661
                           [03]: KB4516115
                           [04]: KB4517245
                           [05]: KB4528759
                           [06]: KB4537759
                           [07]: KB4541338
                           [08]: KB4552152
                           [09]: KB4560959
                           [10]: KB4561600
                           [11]: KB4560960
Network Card(s):           4 NIC(s) Installed.
                           [01]: Intel(R) Wireless-AC 9462
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.114
                                 [02]: fe80::d5ce:74b:f58:55a3
                           [02]: Realtek PCIe GbE Family Controller
                                 Connection Name: Ethernet
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.113
                                 [02]: fe80::d8e4:a052:4f11:d168
                           [03]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [04]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.9.193
                                 [02]: fe80::534:ff05:7329:2dd3
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. Install crc on Windows 10 Pro Workstation
  2. Login to kubeadmin via WebConsole then navigate to Operator Hub
  3. No operators functionality rendered

### Expected
 
See ability to search Operator Hub to located community operators. 

### Actual
**No OperatorHub Items Found**
Please check that the OperatorHub is running and that you have created a valid OperatorSource. For more information about OperatorHub, please click here.
![OperatorHub](https://qas-shared-assets.s3.amazonaws.com/snaps/u6syxitsx9)

**Installed Operators Page**

It states the following:

> Package Server
> 0.14.2 provided by Red Hat
> > Status
> > Cannot update
> > Catalog source was removed

![Installed Operators](https://qas-shared-assets.s3.amazonaws.com/snaps/zt4x2w07tlg)
![Operator Details 1.0](https://qas-shared-assets.s3.amazonaws.com/snaps/hbyhkcis50a)
![Operator Details 1.1](https://qas-shared-assets.s3.amazonaws.com/snaps/4jmc2x7ev79)

**Home > Overview > Cluster Overview**
![Overview - Red Hat Openshift Container Platform](https://qas-shared-assets.s3.amazonaws.com/snaps/nsmzlgdcrlp)

### Logs
https://gist.github.com/GrooveCS/8f2495ce507800814dbda8833f7b23fe#file-crc-2020-05-09t15-43-27-04-00-log

**Comment by praveenkumar** [comment]
@GrooveCS how long have you waited (sometime it takes around 4-5 mins for all the operators appears). In the cli can you run the following command and let us know the output?

```
$ oc get opsrc -n openshift-marketplace
```

**Comment by dgolovin** [comment]
@GrooveCS @praveenkumar I have the same problem, unless I use `-n 1.1.1.1` or `-n 8.8.8.8` when I start crc. DNS can be saved using `crc config set nameserver 8.8.8.8` to avoid adding `-n` every time you start CRC.

**Comment by gbraad** [comment]
This would mean it is an issue with resolving the external address as you would be forcing a nameserver to be used within the VM. This would usually only be a problem on Hyperkit... or Windows 10 Enterprise setups...

It seems you aren't running a regular "Windows 10 Pro" either, but "Microsoft Windows 10 Pro for Workstations" which might explain the behaviour as being more like Enterprise. The use of the Default Switch on that has been reported more oftne than once to be flaky or different.

**Comment by gbraad** [comment]
Can you provide the results of:

```
PS> get-DNSClientServerAddress
```

and from the VM (after using ssh with core@ and the key in the ~/.crc/machines/crc folder):

```
$ cat /etc/resolv.conf
```

**Comment by dgolovin** [comment]
@gbraad I cannot replicate that anymore on crc 1.11.0. It works without `-n` option.

**Comment by gbraad** [comment]
> It works without -n option.

It should work without, so that is good... but before, was that on a different network or with VPN?

**Comment by dgolovin** [comment]
@gbraad the same workstation no network changes. Just used newer CRC release 1.11.0.

**Comment by gbraad** [comment]
There aren't any relevant  ode changes between these releases, beside a slightly newer openshift. Hmmm...

**Comment by GrooveCS** [comment]
Good afternoon @gbraad, @dgolovin & @praveenkumar,

I choose to request the system upgrade from 4.5 to 4.6, and it took several hours (maybe days). I walked away, and started checking what's going on now. 

```!powershell
┌[Parris@com.RedHat]
└[~\CRC\1.11]> .\crc status
CRC VM:          Running
OpenShift:       Running (v4.4.5)
Disk Usage:      16.28GB of 32.72GB (Inside the CRC VM)
Cache Usage:     40.95GB
Cache Directory: C:\Users\Developer\.crc\cache
```

```!powershell
┌[Parris@com.RedHat]
└[~\CRC\1.11]> oc get opsrc -n openshift-marketplace
NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS      MESSAGE                                       AGE
certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Succeeded   The object has been successfully reconciled   23d
community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Succeeded   The object has been successfully reconciled   23d
redhat-marketplace    appregistry   https://quay.io/cnr   redhat-marketplace    Red Hat Marketplace   Red Hat     Succeeded   The object has been successfully reconciled   23d
redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Succeeded   The object has been successfully reconciled   23d
```

I will note to consider to set `crc config set nameserver 8.8.8.8` in the future, yet today it appears not needed. 

Here is what I currently have when run `PS> Get-DnsClientServerAddress`

```!powershell
┌[Parris@com.RedHat]
└[~\CRC\1.11]> Get-DnsClientServerAddress

InterfaceAlias               Interface Address ServerAddresses
--------------               --------- ------- ---------------
Ethernet                             9 IPv4    {192.168.1.1}
Ethernet                             9 IPv6    {}
Wi-Fi                               18 IPv4    {192.168.1.1}
Wi-Fi                               18 IPv6    {}
Local Area Connection* 9            16 IPv4    {}
Local Area Connection* 9            16 IPv6    {fec0:0:0:ffff::1, fec0:0:0:ffff::2, fec0:0:0:ffff::3}
Local Area Connection* 10           11 IPv4    {}
Local Area Connection* 10           11 IPv6    {fec0:0:0:ffff::1, fec0:0:0:ffff::2, fec0:0:0:ffff::3}
Bluetooth Network Connection         7 IPv4    {}
Bluetooth Network Connection         7 IPv6    {fec0:0:0:ffff::1, fec0:0:0:ffff::2, fec0:0:0:ffff::3}
Loopback Pseudo-Interface 1          1 IPv4    {}
Loopback Pseudo-Interface 1          1 IPv6    {fec0:0:0:ffff::1, fec0:0:0:ffff::2, fec0:0:0:ffff::3}
vEthernet (Default Switch)          22 IPv4    {172.17.197.138}
vEthernet (Default Switch)          22 IPv6    {}
```

The Operator page is currently rendering this afternoon. See screenshot. 

## Operator 
![Operator Hub Page](https://qas-shared-assets.s3.amazonaws.com/snaps/g4axr82ytc4)

It currently timing out the first time I access it, yet if I refresh it works after the third or fourth time.  
I updated to candidate-4.4 from stable, and it rendering Operator page. I have not tested the page to see if functional. 

## Overview 
![Overview Page](https://qas-shared-assets.s3.amazonaws.com/snaps/cp0oqi93she)

## Cluster Settings
![Cluster Page](https://qas-shared-assets.s3.amazonaws.com/snaps/p9igzvg3sl)

Let me know if you require any other information to assist with checking the functionality. Thank you for your support in addressing my concern.

**Comment by lmsurpre** [comment]
I have a fresh install of crc on Mac OSX and I'm seeing the same behavior described in this issue.  Everything is up and running but "No OperatorHub Items Found" from the OperatorHub view of the console.
```sh
> crc version
CodeReady Containers version: 1.12.0+6710aff
OpenShift version: 4.4.8 (embedded in binary)
```

I tried setting the nameserver to 8.8.8.8 but it didn't help at all.
Let me know if you'd like me to open it as a new issue or if there's any information I can provide to help with this one.

```sh
> oc get opsrc -n openshift-marketplace
NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS      MESSAGE                                       AGE
certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Succeeded   The object has been successfully reconciled   5d3h
community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Succeeded   The object has been successfully reconciled   5d3h
redhat-marketplace    appregistry   https://quay.io/cnr   redhat-marketplace    Red Hat Marketplace   Red Hat     Succeeded   The object has been successfully reconciled   5d3h
redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Succeeded   The object has been successfully reconciled   5d3h
```

**Comment by lmsurpre** [comment]
I deleted my crc environment and recreated it and now I can see the items under OperatorHub.

**Comment by GrooveCS** [comment]
Hi @gbraad  & @praveenkumar 

Uninstalled `CRC 1.11` and installed `CRC 1.12` and configured CPUs 4 & Memory 32GB. It promptly showed up since I increased the Memory from 9GB. 
**Operator Hub**
![Operator Hub State](https://qas-shared-assets.s3.amazonaws.com/snaps/oigood61j2b)
